zio-http/shared/src/main/scala/zio/http/ZClientAspect.scala

```diff
@@ -492,4 +492,104 @@ object ZClientAspect {
           },
         )
     }
+
+  /**
+   * Client aspect that logs details of web request as curl command
+   */
+  final def curlLogger(
+    verbose: Boolean = true,
+    logEffect: String => UIO[Unit] = (m: String) => ZIO.log(m),
+  )(implicit trace: Trace): ZClientAspect[Nothing, Any, Nothing, Body, Nothing, Any, Nothing, Response] =
+    new ZClientAspect[Nothing, Any, Nothing, Body, Nothing, Any, Nothing, Response] {
+
+      def formatCurlCommand(
```

Comment:
@landlockedsurfer Do you think we could put this capacity to transform a Request to a `curl` String as a method of `Request` and use this method here?

That'd allow people to "toString" their requests in a nice way if they need to (to log it at any moment, for example)

Reply:
@guizmaii Yes! That's definitely a good idea and very useful! I've added the method `zio.http.ZClientAspect.CurlLogger.formatCurlCommand` here:
https://github.com/landlockedsurfer/zio-http/blob/curl-logger/zio-http/shared/src/main/scala/zio/http/ZClientAspect.scala#L600